client/src/providers/AuthContext.js

```diff
@@ -54,7 +55,7 @@ export const useAuth = () => {
       authenticated: user !== null,
       updateUser
     },
-    data: { user, edit }
+    data: { user, edit, history }
```

Comment:
no need to pass history out nor back into the functions like handleRegister you can use it directly inside useAuth just like your [user, setUser] = useState(null) hook, etc...